genie-web/src/main/java/com/netflix/genie/web/data/services/impl/jpa/JpaPersistenceServiceImpl.java

```diff
@@ -2021,6 +2022,72 @@ public JobInfoAggregate getHostJobInformation(@NotBlank final String hostname) {
             updated
         );
     }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void updateRequestedLauncherExt(
+        @NotBlank(message = "No job id entered. Unable to update.") final String id,
+        @NotNull(message = "Status cannot be null.") final JsonNode launcherExtension
+    ) throws NotFoundException {
+        log.debug("[updateRequestedLauncherExt] Requested to update launcher requested ext of job {}", id);
+
+        this.jobRepository
+            .findByUniqueId(id)
+            .orElseThrow(() -> new NotFoundException("No job exists for the id specified"))
+            .setRequestedLauncherExt(launcherExtension);
+
+        log.debug("[updateRequestedLauncherExt] Updated launcher requested ext of job {}", id);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Transactional(readOnly = true)
+    public Optional<JsonNode> getRequestedLauncherExt(@NotBlank final String id) throws NotFoundException {
+        log.debug(
+            "[getRequestedLauncherExt] Requested for job {}", id);
+        return this.jobRepository
+            .findByUniqueId(id)
+            .orElseThrow(() -> new NotFoundException("No job exists for the id specified"))
+            .getRequestedLauncherExt();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void updateLauncherExt(
+        @NotBlank(message = "No job id entered. Unable to update.") final String id,
+        @NotNull(message = "Status cannot be null.") final JsonNode launcherExtension
+    ) throws NotFoundException {
+        log.debug("[updateRequestedLauncherExt] Requested to update launcher ext of job {}", id);
+
+        this.jobRepository
+            .findByUniqueId(id)
+            .orElseThrow(() -> new NotFoundException("No job exists for the id specified"))
+            .setLauncherExt(launcherExtension);
+
+        log.debug("[updateRequestedLauncherExt] Updated launcher ext of job {}", id);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Transactional(readOnly = true)
+    public Optional<JsonNode> getLauncherExt(@NotBlank final String id) throws NotFoundException {
+        log.debug(
+            "[getLauncherExt] Requested for job {}", id);
+        return this.jobRepository
+            .findByUniqueId(id)
+            .orElseThrow(() -> new NotFoundException("No job exists for the id specified"))
+            .getLauncherExt();
```

Comment:
Doing it in this manner will load more data than you likely want as you're reusing the generic `findByUniqueId` method rather than just projecting out the only field you want.

Reply:
Ok, I didn't realize, i duplicated this from other persistence that also loads a single field.
I'll update both to just pull the necessary.